build/templates/matchers.py.mako

```diff
@@ -0,0 +1,194 @@
+'''Matcher classes used by unit tests in order to set mock expectations.
+These work well with our visatype definitions.
+'''
+
+import ctypes
+from ${template_parameters['metadata'].config['module_name']} import visatype
+
+# Base classes
+
+
+class _ScalarMatcher(object):
+    def __init__(self, expected_type, expected_value):
+        self.expected_type = expected_type
+        self.expected_value = expected_value
+
+    def __repr__(self):
```

Comment:
Should all of the matchers have a `__repr__` or is the default good enough for the rest?

Reply:
Now that I see it, that __repr__ implementation is actually outdated. I lean towards removing it, errors were easy enough to follow.